rust/bindings/build.rs

```diff
@@ -0,0 +1,260 @@
+        }
+
+        #space
+    })
+}
+
+struct Method {
+    name: String,
+    result: Option<typify::TypeId>,
+    params: Vec<(String, typify::TypeId)>,
+}
+
+fn extract(s: &str) -> anyhow::Result<&str> {
+    let (_whole, grp) = lazy_regex::regex_captures!("^#/components/schemas/(.*)", s)
+        .context(format!("unsupported reference format in `{}`", s))?;
+    Ok(grp)
+}
+
+/// OpenRPC uses `#/components/schemas/Foo`,
+/// but JSON Schema (and [typify]) uses `#/definitions/Foo`.
+fn rewrite_references(
+    schema: &mut Schema,
+    mut rewrite: impl FnMut(&mut String) -> anyhow::Result<()>,
+) -> anyhow::Result<()> {
+    if let Schema::Object(SchemaObject {
+        reference,
+        subschemas,
+        array,
+        object,
+        ..
+    }) = schema
+    {
+        reference.as_mut().map(cast(&mut rewrite)).transpose()?;
+
+        for child in array
+            .as_deref_mut()
+            .iter_mut()
+            .flat_map(
+                |ArrayValidation {
+                     items,
+                     additional_items,
+                     contains,
+                     ..
+                 }| {
+                    [additional_items, contains]
+                        .into_iter()
+                        .flat_map(Option::as_deref_mut)
+                        .chain(items.iter_mut().flat_map(|it| match it {
+                            SingleOrVec::Vec(it) => &mut it[..],
+                            SingleOrVec::Single(it) => std::slice::from_mut(&mut **it),
+                        }))
+                },
+            )
+            .chain(object.as_deref_mut().iter_mut().flat_map(
+                |ObjectValidation {
+                     properties,
+                     pattern_properties,
+                     additional_properties,
+                     property_names,
+                     ..
+                 }| {
+                    [additional_properties, property_names]
+                        .into_iter()
+                        .flat_map(Option::as_deref_mut)
+                        .chain(properties.values_mut())
+                        .chain(pattern_properties.values_mut())
+                },
+            ))
+            .chain(subschemas.as_deref_mut().into_iter().flat_map(
+                |SubschemaValidation {
+                     all_of,
+                     any_of,
+                     one_of,
+                     not,
+                     if_schema,
+                     then_schema,
+                     else_schema,
+                 }| {
+                    [not, if_schema, then_schema, else_schema]
+                        .into_iter()
+                        .flat_map(Option::as_deref_mut)
+                        .chain([all_of, any_of, one_of].into_iter().flatten().flatten())
+                },
+            ))
+        {
+            rewrite_references(child, cast(&mut rewrite))?;
+        }
+    }
+    Ok(())
+}
+
+// break trait solver cycles
```

Comment:
Would be great to see a proper comment here:
1. What the problem is.
2. How does this solve the problem.